src/analysis/processing/qgsalgorithmrandompointsonlines.cpp

```diff
@@ -0,0 +1,279 @@
+                      "<p>Output from the algorithm:</p> "
+                      "<ul> "
+                      "<li> A point layer containing the random points (<code>OUTPUT</code>).</li> "
+                      "<li> The number of generated features (<code>POINTS_GENERATED</code>).</li> "
+                      "<li> The number of missed points (<code>POINTS_MISSED</code>).</li> "
+                      "<li> The number of features with non-empty geometry and missing points (<code>LINES_WITH_MISSED_POINTS</code>).</li> "
+                      "<li> The number of features with an empty or no geometry (<code>LINES_WITH_EMPTY_OR_NO_GEOMETRY</code>).</li> "
+                      "</ul>"
+                    );
+}
+
+
+QgsRandomPointsOnLinesAlgorithm *QgsRandomPointsOnLinesAlgorithm::createInstance() const
+{
+  return new QgsRandomPointsOnLinesAlgorithm();
+}
+
+bool QgsRandomPointsOnLinesAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  mNumPoints = parameterAsInt( parameters, POINTS_NUMBER, context );
+  mMinDistance = parameterAsDouble( parameters, MIN_DISTANCE, context );
+  mMaxAttempts = parameterAsInt( parameters, MAX_TRIES_PER_POINT, context );
+  mRandSeed = parameterAsInt( parameters, SEED, context );
+  mIncludeLineAttr = parameterAsBoolean( parameters, INCLUDE_LINE_ATTRIBUTES, context );
+  return true;
+}
+
+QVariantMap QgsRandomPointsOnLinesAlgorithm::processAlgorithm( const QVariantMap &parameters,
+    QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr< QgsFeatureSource > lineSource( parameterAsSource( parameters, INPUT, context ) );
+  if ( !lineSource )
+    throw QgsProcessingException( invalidSourceError( parameters, INPUT ) );
+
+  QgsFields fields;
+  fields.append( QgsField( QStringLiteral( "rand_point_id" ), QVariant::LongLong ) );
+  if ( mIncludeLineAttr )
+    fields.extend( lineSource->fields() );
+
+  QString ldest;
+  std::unique_ptr< QgsFeatureSink > sink( parameterAsSink( parameters, OUTPUT,
+                                          context, ldest, fields, QgsWkbTypes::Point, lineSource->sourceCrs() ) );
+  if ( !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, OUTPUT ) );
+
+  //initialize random engine
+  srand( mRandSeed );
+
+  //index for finding close points (mMinDistance > 0)
+  QgsSpatialIndex index;
+
+  int totNPoints = 0;
+  int missedPoints = 0;
+  int missedLines = 0;
+  int emptyOrNullGeom = 0;
+  long tries = 0;
+  long saved = 0;
+
+  long featureCount = 0;
+  long numberOfFeatures = lineSource->featureCount();
+  QgsFeature lFeat;
+  QgsFeatureIterator fitL = mIncludeLineAttr ? lineSource->getFeatures()
+                            : lineSource->getFeatures( QgsFeatureRequest().setNoAttributes() );
+  while ( fitL.nextFeature( lFeat ) )
+  {
+    featureCount++;
+    if ( feedback->isCanceled() )
+    {
+      feedback->pushInfo( "canceled" );
```

Comment:
Can you remove this?

Reply:
Done. Must have forgotten this one... Thanks for pointing it out.